fixes #46 -  structuring and documenting data (#51)
S-ENDA/DMH: documentation/struct_dataset.md

## Dataset

> The dataset is the basic building block of the data management model; all the documentation and services described in this DMH are built on datasets. The dataset specification is the first step in structuring data for efficient management, and it is mandatory.

Reviewer: ```suggestion
Datasets form the building block of any data management model. Dataset specification is the first step in structuring data for efficient management.
```

Reviewer: I think this should come after 

> A dataset is a collection of data. Typically, a dataset represents a number of variables in time and space. Datasets may be categorised by:
